### PR TITLE
Added Deprecation policy to Mantid documentation

### LINKED ISSUE #33792: Create and document an algorithm deprecation/deletion policy
**Describe the outcome that is desired.**

A document in the user documentation describing the process that an algorithm goes through from
being deprecated to being deleted.

It should include a timeline that uses language that users will understand, e.g by release and not sprints.

**Describe any solutions you are considering**
The solution should be a documentation page in the user documentation, clearly linked from the main page.

### PR DESCRIPTION
**Pull request by sf1919**
### Description of work
At the 2023 Mantid Developer meeting a draft deprecation policy was produced (see https://github.com/mantidproject/workshops/blob/main/developer/2023-10/codecamp/deprecation_policy.md) . In July 2024 the Technical Working Group reviewed the draft and this is the final version based on those discussions.

The purpose of the deprecation policy is to set out how we deprecate algorithms, features etc. in Mantid. It covers
- How we determine what should be deprecated
- The timeline for deprecation
- How we will communicate deprecations

It is aimed at both users and developers and as such I have stored this in user documentation to make it visible more widely.

#### Summary of work
A written deprecation policy added to documentation based on this draft - https://github.com/mantidproject/workshops/blob/main/developer/2023-10/codecamp/deprecation_policy.md

Fixes #33792 

### To test:
- Review the documentation and check it covers what is needed, makes sense and spelling and grammar is correct.
- Check the documentation builds without failure and links work
- Check link works in release note

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Review by peterfpeterson — COMMENTED** [review]
Looking at the [governance](https://github.com/mantidproject/governance/) document, maybe we change "facilities" to "Mantid Steering Committee" (with link). Similar to deprecating unused algorithms, we should give a final approval to the steering committee. This is consistent with what we've been doing this last round.

File: docs/source/deprecation.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+------------
+
+This policy lays out how we deprecate algorithms, interfaces and other parts of Mantid. It is intended as a reference
+for both users and developers.
+
+
+What will we deprecate?
+-----------------------
+
+- Algorithms that have not been used for 3 years or more. We will use Usage Data to determine this. For more information
```
[the remark this answers is not in the document]
Comment (sf1919) [reply docs/source/deprecation.rst:20]: See revised lines 27-28 below

**Review by peterfpeterson — DISMISSED** [review]

File: docs/source/deprecation.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  about how we record Usage data please see our `Privacy Policy: Usage Data Recorded in Mantid
+  <https://www.mantidproject.org/privacy.html#usage-data-recorded-in-mantid>`_
+- Other Features covered by Usage Data that have not need used for 3 or more years
```
Comment (robertapplin) [inline docs/source/deprecation.rst:23]:
```suggestion
- Other Features covered by Usage Data that have not beed used for 3 or more years
```

File: docs/source/deprecation.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+- Check usage reporting as to whether algorithm/feature has been used within last 3 years.
+- If it has not be used within last three years make a request to the Technical Working Group (TWG) to have it deprecated
+- If approved by the TWG follow the guidance under the Communication section above.
+
+Soft Deprecation
+----------------
+
+When algorithms are replaced with newer versions, the previous version(s) will be subject to a soft deprecation. We will
+make it clear that we are not longer supporting (i.e. no longer developing) the previous version(s). Although previous
```
Comment (robertapplin) [inline docs/source/deprecation.rst:61]:
```suggestion
make it clear that we are no longer supporting (i.e. no longer developing) the previous version(s). Although previous
```

File: docs/source/deprecation.rst [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+versions will still be available and safe to use users will be encouraged to use more recent versions.
```
Comment (robertapplin) [inline docs/source/deprecation.rst:62]:
```suggestion
versions will still be available and safe to use, users will be encouraged to use more recent versions.
```

**Review by robertapplin — CHANGES_REQUESTED** [review]

**Comment by sf1919** [comment]
I have updated the PR to take into account @robertapplin suggestions. I have also added a little bit about best practice for deprecating multiple algorithms/features as separate PRs.

@peterfpeterson Can you review again?

**Review by peterfpeterson — APPROVED** [review]

**Review by robertapplin — APPROVED** [review]

**Pull request merged**